src/upgrader.ts

```diff
@@ -318,7 +334,7 @@ export class DefaultUpgrader extends EventEmitter<UpgraderEvents> implements Upg
       maConn,
       upgradedConn,
       muxerFactory,
-      remotePeer
+      remotePeer: remotePeerId
```

Comment:
This value used to come from the connection encrypter's `encryptOutbound` return value.  Looks like now that's ignored in favour of the one extracted from the multiaddr.

This will be problematic because we want to support discovering the peer id during the noise handshake in future - please can you change it back to use the one from the connection encrypter if it's configured & returned.